src/schema/v2/me/myCollectionCreateArtworkMutation.ts

```diff
@@ -33,47 +30,36 @@ export const myCollectionCreateArtworkMutation = mutationWithClientMutationId<
     },
   },
   outputFields: {
-    artwork: {
-      type: ArtworkType,
-      resolve: ({ id }, _, { myCollectionArtworkLoader }) => {
-        if (myCollectionArtworkLoader) {
-          return myCollectionArtworkLoader(id)
-        }
-      },
-    },
-    artworkEdge: {
-      type: MyCollectionEdgeType,
-      resolve: async ({ id }, _, { myCollectionArtworkLoader }) => {
-        if (!myCollectionArtworkLoader) {
-          return null
-        }
-        const artwork = await myCollectionArtworkLoader(id)
-        const edge = {
-          cursor: cursorForObjectInConnection([artwork], artwork),
-          node: artwork,
-        }
-        return edge
-      },
+    artworkOrError: {
+      type: MyCollectionArtworkMutationType,
+      resolve: (result) => result,
     },
   },
-  mutateAndGetPayload: (
+  mutateAndGetPayload: async (
     { artistIds, dimensions, medium, title, year },
     { myCollectionCreateArtworkLoader }
   ) => {
     if (!myCollectionCreateArtworkLoader) {
       return new Error("You need to be signed in to perform this action")
     }
 
-    return myCollectionCreateArtworkLoader({
-      artist_ids: artistIds,
-      dimensions,
-      medium,
-      title,
-      year,
-    }).then(({ id }) => {
-      return {
-        id,
+    try {
+      const response = await myCollectionCreateArtworkLoader({
+        artist_ids: artistIds,
+        dimensions,
+        medium,
+        title,
+        year,
+      })
+
+      return response
+    } catch (error) {
+      const formattedErr = formatGravityError(error)
+      if (formattedErr) {
+        return { ...formattedErr, _type: "GravityMutationError" }
+      } else {
+        throw new Error(error)
```

Comment:
**question [potentially blocking]:**

Is there anything to be worried about with throwing a generic `Error` here, given that [the Graphql type wrapping these errors](https://github.com/artsy/metaphysics/pull/2654/files#diff-8d4fff9e1cab233bb45d1369c1325a66R133) seems to assume that any underlying errors are `GravityMutationError`s? 

_Comment format based on [conventionalcomments.org](https://conventionalcomments.org/)_

Reply:
This is 🍝 which matches the other examples in codebase. What happens under the hood is if there's no match for that type then it will throw at a lower level and return an exception, versus the graceful handling setup via this design pattern.

Reply:
Looking at this, we could probably wrap this error stuff in a reusable function and DRY up the codebase more. I can do that in a follow-up.

Reply:
No rush on my end for that follow-up, I mostly just wanted to make sure I understood the code that was there.